test/test_parentwork.py

```diff
@@ -85,14 +132,62 @@ def test_no_force(self):
     @unittest.skipUnless(
         os.environ.get('INTEGRATION_TEST', '0') == '1',
         'integration testing not enabled')
-    def test_direct_parent_work(self):
+    def test_direct_parent_work_real(self):
         mb_workid = u'2e4a3668-458d-3b2a-8be2-0b08e0d8243a'
         self.assertEqual(u'f04b42df-7251-4d86-a5ee-67cfa49580d1',
                          parentwork.direct_parent_id(mb_workid)[0])
         self.assertEqual(u'45afb3b2-18ac-4187-bc72-beb1b1c194ba',
                          parentwork.work_parent_id(mb_workid)[0])
 
 
+class ParentWorkMockTest(unittest.TestCase, TestHelper):
+    def setUp(self):
+        """Set up configuration"""
+        self.setup_beets()
+        self.load_plugins('parentwork')
+
+    def tearDown(self):
+        self.unload_plugins()
+        self.teardown_beets()
+
+    musicbrainzngs.get_work_by_id = Mock(side_effect=mock_workid_response)
```

Comment:
Doing this mocking "in class scope" will, unfortunately, *permanently* mock the function as soon as the class is loaded. Instead, there are a couple of options to "temporarily" mocking something:

- Use `patch` in `setUp` and `tearDown` as described in the `mock` docs: https://docs.python.org/3/library/unittest.mock.html#patch-methods-start-and-stop
- Use the `@patch` decorator: https://stackoverflow.com/a/23618258/39182
- Manually save the "old" `get_work_by_id` in `setUp` and restore it in `tearDown`.

Reply:
Ok I'll try it tomorrow (it's midnight here)

Reply:
if the class-level is the problem, could it be mocked at method-level?

Reply:
About your first example: if I use ```patch``` in ```setUp``` and ```tearDown``` , where do I control in which method I use the mock and in which I use the real one? I can't find it explained in the ```mock``` docs. 
I thought about this "hybrid" mocking, where I mock the functions for the arguments I use for the mock tests and I just keep the real function for all other arguments (like here https://github.com/beetbox/beets/pull/3671/commits/dba2e2320da48a653aaebcc7cb61b982fe867f99), could that be an option?

Reply:
If you set something up in `setUp` and tear it down in `tearDown`, it will be "in effect" for all test invocations in that class. If you need a test that does not have that fixture, you can put it in a different class (such as the two classes you have currently defined).

I think that's likely to be simpler/easier to predict than deciding whether to do "mock" or "real" requests based on the argument…

Reply:
Does it correspond more to what you have in mind now @sampsyo ?